### PR TITLE
BE-1396 first use of event api

### PR DESCRIPTION
**Pull request by josecordaz**
**JIRA:** https://pinpt-hq.atlassian.net/browse/BE-1396

**Description:**

### CODE REVIEW

File: pkg/upload/upload.go [hunk 1]
```diff
@@ -488,15 +489,41 @@ type Start struct {
...
+	return agent.PostEvent(ctx, event, channel, apiKey, nil)
+}
+
+func appendCommonInfo(event *agent.Event) {
+
+	customerID := integration.GetCustomerID()
+	systemInfo := sysinfo.GetSystemInfo()
+
+	(*event).ID = hash.Values("Event", customerID, "export", "Not sure")
+	(*event).RefID = "not sure"
```
Comment (jhaynie) [inline pkg/upload/upload.go]: let the system hash the id
keep ref_id blank
you never should set hashcode (even empty, the system generates this)

File: pkg/upload/upload.go [hunk 2]
```diff
@@ -488,15 +489,41 @@ type Start struct {
...
-	start := Start{
-		SystemInfo: sysinfo.GetSystemInfo(),
-		Date:       datetime.EpochNow(),
-		CustomerID: customerID,
+func SendStart(ctx context.Context, logger log.Logger, channel string, apiKey string) error {
+
+	event := agent.Event{
+		RefType: "export",
+		Type:    "start",
+		Distro:  "not sure",
```
Comment (jhaynie) [inline pkg/upload/upload.go]: the default for strings is empty so you never need to set them explictly to empty string
for distro, this is the distribution which you should be able to get from the old code

File: pkg/upload/upload.go [hunk 3]
```diff
@@ -488,15 +489,41 @@ type Start struct {
...
+	(*event).OS = systemInfo.OS
+	(*event).Version = systemInfo.Version
+	(*event).Hostname = systemInfo.Hostname
+	(*event).NumCPU = int64(systemInfo.NumCPU)
+	(*event).FreeSpace = int64(systemInfo.FreeSpace)
+	(*event).GoVersion = systemInfo.GoVersion
+	(*event).Architecture = systemInfo.Architecture
+	(*event).Memory = int64(systemInfo.Memory)
+	(*event).UUID = getInstallUUID()
+	(*event).Date = datetime.ISODateFromTime(time.Now())
```
Comment (jhaynie) [inline pkg/upload/upload.go]: use RFC3339 time instead of UTC

**Comment by josecordaz** [comment]
> seems like you've just done one of them? comments inline

yes

File: pkg/upload/upload.go [hunk 4]
```diff
@@ -510,19 +508,18 @@ func appendCommonInfo(event *agent.Event) {
 	systemInfo := sysinfo.GetSystemInfo()
 
 	(*event).ID = hash.Values("Event", customerID, "export", "Not sure")
-	(*event).RefID = "not sure"
 	(*event).CustomerID = customerID
-	(*event).Hashcode = ""
 	(*event).OS = systemInfo.OS
 	(*event).Version = systemInfo.Version
 	(*event).Hostname = systemInfo.Hostname
+	(*event).Distro = systemInfo.Name + "_" + systemInfo.Version
```
Comment (jhaynie) [inline pkg/upload/upload.go]: 👍

File: cmd/onboard.go [hunk 5]
```diff
@@ -118,7 +118,7 @@ func sendProgress(ctx context.Context, logger log.Logger, mainProgress *progress
 	for {
 		select {
 		case <-time.After(time.Second * 1):
```
Comment (jhaynie) [inline cmd/onboard.go]: we need to re-write the entire onboard process next week to use the event api subscriptions. we also need to add the subscription interface to go-common/event as well so you can then use it. we should merge this though and do all that in a separate PR since it will be a big change.
